cub/cub/device/dispatch/dispatch_reduce_deterministic.cuh

```diff
@@ -0,0 +1,502 @@
+    using MaxPolicyT        = typename PolicyHub::MaxPolicy;
+
+    // Force kernel code-generation in all compiler passes
+    if (num_items <= (SingleTilePolicyT::BLOCK_THREADS * SingleTilePolicyT::ITEMS_PER_THREAD))
+    {
+      return InvokeSingleTile<ActivePolicyT>(
+        detail::reduce::DeterministicDeviceReduceSingleTileKernel<
+          MaxPolicyT,
+          InputIteratorT,
+          OutputIteratorTransformT,
+          OffsetT,
+          ReductionOpT,
+          InitT,
+          deterministic_accum_t,
+          TransformOpT>);
+    }
+    else
+    {
+      return InvokePasses<ActivePolicyT>(
+        detail::reduce::DeterministicDeviceReduceKernel<
+          MaxPolicyT,
+          InputIteratorT,
+          OffsetT,
+          ReductionOpT,
+          deterministic_accum_t,
+          TransformOpT>,
+        detail::reduce::DeterministicDeviceReduceSingleTileKernel<
+          MaxPolicyT,
+          deterministic_accum_t*,
+          OutputIteratorTransformT,
+          int, // Always used with int offsets
+          ReductionOpT,
+          InitT,
+          deterministic_accum_t>);
+    }
+  }
+
+  //---------------------------------------------------------------------------
+  // Dispatch entrypoints
+  //---------------------------------------------------------------------------
+
+  /**
+   * @brief Internal dispatch routine for computing a device-wide deterministic reduction
+   *
+   * @param[in] d_temp_storage
+   *   Device-accessible allocation of temporary storage. When `nullptr`, the
+   *   required allocation size is written to `temp_storage_bytes` and no work
+   *   is done.
+   *
+   * @param[in,out] temp_storage_bytes
+   *   Reference to size in bytes of `d_temp_storage` allocation
+   *
+   * @param[in] d_in
+   *   Pointer to the input sequence of data items
+   *
+   * @param[out] d_out
+   *   Pointer to the output aggregate
+   *
+   * @param[in] num_items
+   *   Total number of input items (i.e., length of `d_in`)
+   *
+   * @param[in] init
+   *   The initial value of the reduction
+   *
+   * @param[in] stream
+   *   **[optional]** CUDA stream to launch kernels within.
+   *   Default is stream<sub>0</sub>.
+   */
+  CUB_RUNTIME_FUNCTION _CCCL_FORCEINLINE static cudaError_t Dispatch(
+    void* d_temp_storage,
+    size_t& temp_storage_bytes,
+    InputIteratorT d_in,
+    OutputIteratorT d_out,
+    OffsetT num_items,
+    rfa::InitT<OutputIteratorT, InputIteratorT> init = {},
+    cudaStream_t stream                              = {},
+    TransformOpT transform_op                        = {})
+  {
+    cudaError error = cudaSuccess;
+
+    // Get PTX version
+    int ptx_version = 0;
+    error           = CubDebug(PtxVersion(ptx_version));
+    if (cudaSuccess != error)
+    {
+      return error;
+    }
+
+    OutputIteratorTransformT d_out_transformed =
+      THRUST_NS_QUALIFIER::make_transform_output_iterator(d_out, AcumFloatTransformT{});
```

Comment:
suggestion: this is a leftover. We now have custom kernels for RFA implementation. Instead of complicating compilation with fancy iterators, we can just convert to float in the RFA kernels upon writing into `d_out`.